src/target/x86.rs

```diff
@@ -0,0 +1,201 @@
+        match slower {
+            X86MemoryLevel::RF | X86MemoryLevel::VRF => vec![],
+            X86MemoryLevel::L1 => vec![X86MemoryLevel::RF, X86MemoryLevel::VRF],
+            X86MemoryLevel::GL => vec![X86MemoryLevel::L1],
+        }
+    }
+
+    fn actions(spec: &LogicalSpec<Self>) -> Box<dyn Iterator<Item = Action<Self>>> {
+        {
+            match spec {
+                LogicalSpec::Primitive(PrimitiveBasics { typ, .. }, _, _) => match typ {
+                    PrimitiveSpecType::Matmul { accum } => {
+                        if *accum {
+                            let mut microkernels = vec![];
+                            if mult_applies_to_operands(&spec.parameters()) {
+                                microkernels.push(Action::Place(KernelType::Mult));
+                            }
+                            if broadcastvecmult_applies_to_operands(&spec.parameters()) {
+                                microkernels.push(Action::Place(KernelType::BroadcastVecMult));
+                            }
+                            Box::new(microkernels.into_iter())
+                        } else {
+                            Box::new(iter::empty())
+                        }
+                    }
+                    PrimitiveSpecType::Conv { .. } => Box::new(iter::empty()),
+                    PrimitiveSpecType::Move { .. } => {
+                        let mut microkernels = vec![];
+                        if valueassign_applies_to_operands(&spec.parameters()) {
+                            microkernels.push(Action::Place(KernelType::ValueAssign));
+                        }
+                        if vectorassign_applies_to_operands(&spec.parameters()) {
+                            microkernels.push(Action::Place(KernelType::VectorAssign));
+                        }
+                        Box::new(microkernels.into_iter())
+                    }
+                    PrimitiveSpecType::Zero { .. } => {
+                        let mut microkernels = vec![];
+                        if memsetzero_applies_to_operands(&spec.parameters()) {
+                            microkernels.push(Action::Place(KernelType::MemsetZero));
+                        }
+                        if vectorzero_applies_to_operands(&spec.parameters()) {
+                            microkernels.push(Action::Place(KernelType::VectorZero));
+                        }
+                        Box::new(microkernels.into_iter())
+                    }
+                },
+                LogicalSpec::Compose { .. } => Box::new(iter::empty()),
+            }
+        }
+    }
+
+    fn by_enum() -> Targets {
+        Targets::X86
+    }
+
+    fn vec_types() -> &'static [VecType; 4] {
+        &X86_VEC_TYPES
+    }
+}
+
+#[allow(clippy::upper_case_acronyms)]
+#[derive(
+    Eq, PartialEq, Debug, Copy, Clone, Hash, Deserialize, Serialize, enum_iterator::Sequence,
+)]
+pub enum X86MemoryLevel {
```

Comment:
Since this is used by both X86Target and ArmTarget, we should rename this to `CpuMemoryLevel` and lift to target/mod.rs.